backend/main.py

```
@@ -0,0 +1,312 @@
+    try:
+        if request.source == "youtube":
+            songs = await search_youtube_tracks(request.query, request.limit)
+        elif request.source == "spotify":
+            songs = await search_spotify_tracks(request.query, request.limit)
+        else:
+            raise HTTPException(status_code=400, detail="Invalid source. Use 'youtube' or 'spotify'")
+        
+        return {"songs": songs}
+    except Exception as e:
+        logger.error(f"Search error: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)